File: src/domain/entities/Product_Image.ts
import Product from "./Product";
import Product_Option from "./Product_Option";

export default class Product_Image {
  constructor(
    public url: string,
    public alt_text: string,
    public is_primary: boolean,
    public display_order: number,
    public product?: Product,
    public option?: Product_Option,
    public product_id?: number,
    public option_id?: number | null,
    public id?: number,
  ) {}
}

File: src/main/java/minseok/cqrschallenge/product/entity/ProductImage.java
package minseok.cqrschallenge.product.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;

@Entity
@Table(name = "product_images")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(nullable = false)
    private String url;

    @Column(name = "alt_text")
    private String altText;

    @Column(name = "is_primary")
    private Boolean isPrimary = false;

    @Column(name = "display_order")
    private Integer displayOrder = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "option_id")
    private ProductOption option;
}